### ISSUE TITLE
Crashing frequently on OS X: Collection was modified; enumeration operation may not execute

### ISSUE THREAD

**Issue by sn4k3** [AvaloniaUI/Avalonia#13124]
## Describe the bug

```
System.InvalidOperationException: Collection was modified; enumeration operation may not execute.
   at Avalonia.Rendering.Composition.Server.ServerCompositor.NotifyBatchesProcessed()
   at Avalonia.Rendering.Composition.Server.ServerCompositor.RenderCore()
   at Avalonia.Rendering.Composition.Server.ServerCompositor.Render()
   at Avalonia.Media.MediaContext.SyncCommit(Compositor compositor, Boolean waitFullRender)
   at Avalonia.Rendering.Composition.CompositingRenderer.Paint(Rect rect)
   at Avalonia.Controls.TopLevel.HandlePaint(Rect rect)
   at Avalonia.Native.WindowBaseImpl.WindowBaseEvents.Avalonia.Native.Interop.IAvnWindowBaseEvents.Paint()
   at Avalonia.Native.Interop.Impl.__MicroComIAvnWindowBaseEventsVTable.Paint(Void* this)
--- End of stack trace from previous location ---
   at Avalonia.Native.DispatcherImpl.RunLoop(CancellationToken token)
   at Avalonia.Threading.DispatcherFrame.Run(IControlledDispatcherImpl impl)
   at Avalonia.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   at Avalonia.Threading.Dispatcher.MainLoop(CancellationToken cancellationToken)
   at Avalonia.Controls.ApplicationLifetimes.ClassicDesktopStyleApplicationLifetime.Start(String[] args)
   at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime(AppBuilder builder, String[] args, ShutdownMode shutdownMode)
   at UVtools.UI.Program.Main(String[] args) in UVtools.UI/Program.cs:line 174
Category: Application

Machine date time: 10/1/2023 10:49:28 AM
    UTC date time: 10/1/2023 2:49:28 PM
```

## To Reproduce

I cannot reproduce, only got the issue from a client that run on macOS: https://github.com/sn4k3/UVtools/issues/762
Client told this happen when checking for issues on my program. Which during the detection task the UI only get's affected by the progress (bar, title, text) and when it finishes a DataGrid source is set with the detected issues.

## Environment 

```
UVtools v4.0.1 Arm64
Operative system: Darwin 22.6.0 Darwin Kernel Version 22.6.0: Wed Jul  5 22:22:05 PDT 2023; root:xnu-8796.141.3~6/RELEASE_ARM64_T6000 Arm64
Processor: Apple M1 Pro
Processor cores: 8
Memory RAM: 5.48 / 10.10 GB
Runtime: osx.13-arm64
Framework: .NET 6.0.22
AvaloniaUI: 11.0.4
```

## Additional Context

So far not a problem when on Windows as far I can test.

**Comment by timunie** [comment]
Can you check with latest nightly? That would be great help. There were some fixes probably related but hard to tell for me as I have no mac on my own

**Comment by sn4k3** [comment]
Me either, Windows only... I have a macOS, let-me check if I can trigger the problem there. If not I can try a nightly compilation and send to user for try. Will report back.

**Comment by maxkatz6** [comment]
This issue was most likely fixed with this PR https://github.com/AvaloniaUI/Avalonia/pull/12968
Soon to be available in 11.0.5
